app/src/main/java/com/amaze/filemanager/ui/fragments/preference_fragments/PrefFrag.java

```diff
@@ -231,12 +241,14 @@ public boolean onPreferenceClick(Preference preference) {
                   sharedPref
                       .edit()
                       .putInt(PreferencesConstants.PREFERENCE_DRAG_AND_DROP_PREFERENCE, which)
+                      .putString(PreferencesConstants.PREFERENCE_DRAG_AND_DROP_SUMMARY, getDragDropSummary(which))
```

Comment:
Why are you adding a new thing to the SharedPreferences?

Reply:
for storing the selected option text in the preference so that when preferences are created again it will display stored selection in summary

Reply:
This is absolutely not necessary. just set the summary based on preference value you're getting from preferences dynamically.

Reply:
we get only the preference id we dont get title for that id so for that I have used a seperate preference constant to store title which I am fetching from that id through getDragDropSummary method